pkg/controllermanager/controller/shoot/shoot_quota_control.go

```diff
@@ -136,12 +128,12 @@ func (r *shootQuotaReconciler) Reconcile(ctx context.Context, request reconcile.
 		r.logger.Info("[SHOOT QUOTA] Shoot cluster lifetime expired. Shoot will be deleted.")
 
 		// We have to annotate the Shoot to confirm the deletion.
-		if err := common.ConfirmDeletion(ctx, gardenClient.Client(), shoot); err != nil {
+		if err := common.ConfirmDeletion(ctx, r.gardenClient.Client(), shoot); err != nil {
 			return reconcile.Result{}, err
 		}
 
 		// Now we are allowed to delete the Shoot (to set the deletionTimestamp).
-		if err := gardenClient.GardenCore().CoreV1beta1().Shoots(shoot.Namespace).Delete(ctx, shoot.Name, metav1.DeleteOptions{}); err != nil {
+		if err := r.gardenClient.GardenCore().CoreV1beta1().Shoots(shoot.Namespace).Delete(ctx, shoot.Name, metav1.DeleteOptions{}); err != nil {
```

Comment:
And also here?
Then we could change the member
`gardenClient kubernetes.Interface`
to `gardenClient client.Client`.